### PR TITLE
refactor(test): move hash agg executor tests to `integration_tests`

### LINKED ISSUE #10005: executor snapshot testing: the ordering of rows in output chunks
Executors like `HashAgg` produces chunks in which the order of rows does not matter, so previously we just sort them before `assert_eq!`. However, some other executors like `Sort` produces rows in order, we may want snapshot tests to detect wrong ordering in such cases.

What do you think? @xxchan

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR also adds `SnapshotOptions` to resolve #10005, and `risedev update-stream-integration-test` subcommand for convenience (mentioned in https://github.com/risingwavelabs/risingwave/pull/9787#issuecomment-1554197182).

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- ~~[ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
- ~~[ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/stream/tests/integration_tests/integration_test.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#!/usr/bin/env bash
+set -e
+
+UPDATE_EXPECT=1 cargo nextest run -p risingwave_stream --test integration_tests --retries 0
+
+echo "$(tput setaf 2)Diff applied!$(tput sgr 0)"
+echo "Tip: use the alias $(tput setaf 4)./risedev usit$(tput sgr0)."
+'''
+
+[tasks.usit]
```
Comment (xxchan) [inline src/stream/tests/integration_tests/integration_test.toml:15]: What about `dasit` or `dast`? similar to `dapt`

File: src/stream/src/executor/hash_agg.rs [hunk 2]
```diff
@@ -670,580 +670,3 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
...
-    async fn test_local_hash_aggregation_min_in_memory() {
-        test_local_hash_aggregation_min(MemoryStateStore::new()).await
-    }
-
-    #[tokio::test]
-    async fn test_local_hash_aggregation_min_append_only_in_memory() {
-        test_local_hash_aggregation_min_append_only(MemoryStateStore::new()).await
-    }
-
-    async fn test_local_hash_aggregation_count<S: StateStore>(store: S) {
```
Comment (xxchan) [inline src/stream/src/executor/hash_agg.rs:712]: This test is deleted. Is that intended?

**Review by xxchan — APPROVED** [review]

File: src/stream/tests/integration_tests/integration_test.toml (hunk 1 above)
Comment (richardchien) [reply src/stream/tests/integration_tests/integration_test.toml:15]: `do-apply-stream-integration-test`?

File: src/stream/src/executor/hash_agg.rs (hunk 2 above)
Comment (richardchien) [reply src/stream/src/executor/hash_agg.rs:712]: Yes, this test looks the same as another so called _global_ one.

**Pull request merged**